src/main.cpp

```diff
@@ -90,24 +91,62 @@ namespace
 
     auto rule_exists(irods::default_re_ctx&, const std::string& _rule_name, bool& _exists) -> irods::error
     {
-        _exists = (_rule_name == "pep_api_mod_avu_metadata_pre");
-        return SUCCESS();
-    }
+		_exists = (_rule_name == "pep_api_mod_avu_metadata_pre") ||
+		          (_rule_name == "pep_api_atomic_apply_metadata_operations_pre");
+		return SUCCESS();
+	}
 
-    auto list_rules(irods::default_re_ctx&, std::vector<std::string>& _rules) -> irods::error
-    {
+	auto list_rules(irods::default_re_ctx&, std::vector<std::string>& _rules) -> irods::error
+	{
         _rules.push_back("pep_api_mod_avu_metadata_pre");
-        return SUCCESS();
-    }
-
-    auto exec_rule(irods::default_re_ctx&,
-                   const std::string& _rule_name,
-                   std::list<boost::any>& _rule_arguments,
-                   irods::callback _effect_handler) -> irods::error
-    {
-        try {
-            auto* input = boost::any_cast<modAVUMetadataInp_t*>(*std::next(std::begin(_rule_arguments), 2));
-            auto& rei = get_rei(_effect_handler);
+		_rules.push_back("pep_api_atomic_apply_metadata_operations_pre");
+		return SUCCESS();
+	}
+
+	auto rule_logic(const std::string& _object_name, ruleExecInfo_t& rei) -> irods::error
```

Comment:
I feel like this name could stand to be a little more detailed. Is it specific to the atomic metadata API (since it's only being used there at the moment), or could it also be useful for the others?

Reply:
Changed the name, but not sure if it's good. My original idea was to make this generic function and eventually make the old function also call this guy (since the logic is effectively the same, just different boilerplate). Taking name suggestions!

Reply:
Looks like there's a name suggestion here: https://github.com/irods/irods_rule_engine_plugin_metadata_guard/pull/65/files#r1463702239

I'll go ahead and resolve this in favor of that thread. Thanks